### PR TITLE
Run3-gem36 Use MuonGeometryConstants rather than MuonDDDConstants

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Use the object MuonGeometryConstants in all the downstream codes rather than MuonDDDConstants

#### PR validation:

Use runTheMatrix.py to validate the tests

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@watson-ij @slomeo @ianna @cvuosalo Trying to introduce MuonGeometryParameters instead of MuonDDDParameters. It affects so many files, I wonder how to sort out conflict with #29609

**Comment by cvuosalo** [comment]
@bsunanda Is it really necessary to do this massive name change? With 
```
using MuonDDDConstants = MuonGeometryConstants;
```
most of the old code could be left untouched. Or am I missing some issue?
`Geometry/MuonNumbering/interface/MuonDDDConstants.h` could be retained, but it would be empty except to include `Geometry/MuonNumbering/interface/MuonGeometryConstants.h`

**Comment by bsunanda** [comment]
@cvuosalo Lots of files just includes one cff file. Also accessing MuonGeometryConstants is from IdealGeometryRecord. So when I needed to make that change I also changed MuonDDDConstants to MuonGeometryConstants. The number of files to be changed will remain roughly the same.

**Comment by cvuosalo** [comment]
@bsunanda OK. It is better to make the full name change, and I guess you have already done all of the work to do it.

**Comment by slomeo** [comment]
@bsunanda  : 
>  I wonder how to sort out conflict with #29609
probably it is better to wait till #29609 will be merged (I hope asap) because your PR will modify only the DD part of it. Please, can  @cvuosalo @ianna @watson-ij comment?

File: Geometry/MuonNumbering/src/MuonDDDNumbering.cc [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
-#include "Geometry/MuonNumbering/interface/MuonDDDConstants.h"
+#include "Geometry/MuonNumbering/interface/MuonGeometryConstants.h"
 #include "FWCore/MessageLogger/interface/MessageLogger.h"
 
 //#define LOCAL_DEBUG
 
-MuonDDDNumbering::MuonDDDNumbering(const MuonDDDConstants &muonConstants) {
-  //  MuonDDDConstants muonConstants;
+MuonDDDNumbering::MuonDDDNumbering(const MuonGeometryConstants &muonConstants) {
+  //  MuonGeometryConstants muonConstants;
```
Comment (kpedro88) [inline Geometry/MuonNumbering/src/MuonDDDNumbering.cc:9]: delete commented-out code

File: SimG4CMS/Muon/src/MuonSensitiveDetector.cc [hunk 2]
```diff
@@ -56,14 +57,17 @@ MuonSensitiveDetector::MuonSensitiveDetector(const std::string& name,
...
 
-  edm::ESTransientHandle<DDCompactView> cpv;
-  es.get<IdealGeometryRecord>().get(cpv);
-
   //The constants take time to calculate and are needed by many helpers
-  MuonDDDConstants constants(*cpv);
+  edm::ESHandle<MuonGeometryConstants> mdc;
+  es.get<IdealGeometryRecord>().get(mdc);
+  if (!mdc.isValid())
+    throw cms::Exception("Unknown", "MuonSensitiveDetector") << "Cannot find MuonGeometryConstants\n";
```
Comment (kpedro88) [inline SimG4CMS/Muon/src/MuonSensitiveDetector.cc:69]: "Unknown" category is unusual, probably just using "MuonSensitiveDetector" as the category is better

**Comment by kpedro88** [comment]
@bsunanda this needs to be rebased now that #29607 is merged (and please make sure to apply your change to the new geometry introduced in that PR)

File: Geometry/MuonNumbering/src/MuonDDDNumbering.cc (hunk 1 above)
Comment (bsunanda) [reply Geometry/MuonNumbering/src/MuonDDDNumbering.cc]: Left a comment. No commented out code there any more

File: SimG4CMS/Muon/src/MuonSensitiveDetector.cc (hunk 2 above)
Comment (bsunanda) [reply SimG4CMS/Muon/src/MuonSensitiveDetector.cc]: Done

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild please test

**Comment by cvuosalo** [comment]
+1

**Comment by kpedro88** [comment]
+upgrade

**Comment by civanch** [comment]
+1

**Comment by pohsun** [comment]
+1

**Comment by silviodonato** [comment]
Do you have any comments @ggovi ?

**Comment by bsunanda** [comment]
@silviodonato Can this be merged at the earliest and then one can concentrate on the other muon related PR's

**Comment by silviodonato** [comment]
merge
@ggovi  please let me know if you have any objections

**Pull request merged**